subprojects/packagefiles/libssh2/example/meson.build

```diff
@@ -0,0 +1,34 @@
+examples_c = [
+    'direct_tcpip.c',
+    'scp.c',
+    'scp_nonblock.c',
+    'scp_write.c',
+    'scp_write_nonblock.c',
+    'sftp.c',
+    'sftp_RW_nonblock.c',
+    'sftp_append.c',
+    'sftp_mkdir.c',
+    'sftp_mkdir_nonblock.c',
+    'sftp_nonblock.c',
+    'sftp_write.c',
+    'sftp_write_nonblock.c',
+    'sftp_write_sliding.c',
+    'sftpdir.c',
+    'sftpdir_nonblock.c',
+    'ssh2.c',
+    'ssh2_agent.c',
+    'ssh2_echo.c',
+    'ssh2_exec.c',
+    'subsystem_netconf.c',
+    'tcpip-forward.c'
+]
+
+x11_dep = dependency('x11')
+
+if x11_dep.found()
```

Comment:
Unless you pass `required: false` to the dependency, it will error out when not found anyway.